kv/kvserver: TestReplicaCircuitBreaker_Partial_Retry failed

kv/kvserver.TestReplicaCircuitBreaker_Partial_Retry [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20520328?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20520328?buildTab=artifacts#/) on release-25.4 @ [b3d1e3ace7ad6ad08bc0d34ef101f06f632cc321](https://github.com/cockroachdb/cockroach/commits/b3d1e3ace7ad6ad08bc0d34ef101f06f632cc321):


```
    client_raft_helpers_test.go:557: dropping StoreLiveness msg MsgHeartbeat from store 2: to 3
    client_raft_helpers_test.go:559: allowing StoreLiveness msg MsgHeartbeatResp from store 1: to 2
    client_raft_helpers_test.go:559: allowing StoreLiveness msg MsgHeartbeat from store 3: to 3
    client_raft_helpers_test.go:559: allowing StoreLiveness msg MsgHeartbeatResp from store 3: to 3
    client_raft_helpers_test.go:557: dropping StoreLiveness msg MsgHeartbeat from store 3: to 2
    client_raft_helpers_test.go:557: dropping StoreLiveness msg MsgHeartbeat from store 3: to 1
    client_replica_circuit_breaker_test.go:947: raft leadership moved to n2
    client_replica_circuit_breaker_test.go:949: writes succeeded
    client_raft_helpers_test.go:543: from store 1; adding store 2 to drop list
    client_raft_helpers_test.go:543: from store 1; adding store 3 to drop list
    client_raft_helpers_test.go:543: from store 2; adding store 1 to drop list
    client_raft_helpers_test.go:543: from store 2; adding store 3 to drop list
    client_replica_circuit_breaker_test.go:958: partitioned all nodes and tripped their breakers
    client_replica_circuit_breaker_test.go:962: writes failed with ReplicaUnavailableError
    client_replica_circuit_breaker_test.go:968: expired n2 lease
    client_replica_circuit_breaker_test.go:970: writes failed with ReplicaUnavailableError
    client_raft_helpers_test.go:559: allowing StoreLiveness msg MsgHeartbeat from store 1: to 1
    client_raft_helpers_test.go:559: allowing StoreLiveness msg MsgHeartbeatResp from store 1: to 1
    client_raft_helpers_test.go:557: dropping StoreLiveness msg MsgHeartbeat from store 1: to 2
    client_raft_helpers_test.go:557: dropping StoreLiveness msg MsgHeartbeat from store 1: to 3
    client_raft_helpers_test.go:559: allowing StoreLiveness msg MsgHeartbeat from store 2: to 2
    client_raft_helpers_test.go:559: allowing StoreLiveness msg MsgHeartbeatResp from store 2: to 2
    client_raft_helpers_test.go:557: dropping StoreLiveness msg MsgHeartbeat from store 2: to 3
    client_raft_helpers_test.go:557: dropping StoreLiveness msg MsgHeartbeat from store 2: to 1
    client_raft_helpers_test.go:559: allowing StoreLiveness msg MsgHeartbeat from store 3: to 3
    client_raft_helpers_test.go:559: allowing StoreLiveness msg MsgHeartbeatResp from store 3: to 3
    client_raft_helpers_test.go:557: dropping StoreLiveness msg MsgHeartbeat from store 3: to 2
    client_raft_helpers_test.go:557: dropping StoreLiveness msg MsgHeartbeat from store 3: to 1
    client_raft_helpers_test.go:559: allowing StoreLiveness msg MsgHeartbeat from store 1: to 1
    client_raft_helpers_test.go:559: allowing StoreLiveness msg MsgHeartbeatResp from store 1: to 1
    client_raft_helpers_test.go:557: dropping StoreLiveness msg MsgHeartbeat from store 1: to 2
    client_raft_helpers_test.go:557: dropping StoreLiveness msg MsgHeartbeat from store 1: to 3
    client_raft_helpers_test.go:559: allowing StoreLiveness msg MsgHeartbeat from store 2: to 2
    client_raft_helpers_test.go:559: allowing StoreLiveness msg MsgHeartbeatResp from store 2: to 2
    client_raft_helpers_test.go:557: dropping StoreLiveness msg MsgHeartbeat from store 2: to 1
    client_raft_helpers_test.go:557: dropping StoreLiveness msg MsgHeartbeat from store 2: to 3
    client_raft_helpers_test.go:559: allowing StoreLiveness msg MsgHeartbeat from store 3: to 3
    client_raft_helpers_test.go:559: allowing StoreLiveness msg MsgHeartbeatResp from store 3: to 3
    client_raft_helpers_test.go:557: dropping StoreLiveness msg MsgHeartbeat from store 3: to 2
    client_raft_helpers_test.go:557: dropping StoreLiveness msg MsgHeartbeat from store 3: to 1
    client_replica_circuit_breaker_test.go:982: no raft leader
    client_replica_circuit_breaker_test.go:984: 
        	Error Trace:	pkg/kv/kvserver/client_replica_circuit_breaker_test.go:867
        	            				pkg/kv/kvserver/client_replica_circuit_breaker_test.go:984
        	            				pkg/testutils/subtest.go:23
        	Error:      	Not equal: 
        	            	expected: int(0)
        	            	actual  : int64(1)
        	Test:       	TestReplicaCircuitBreaker_Partial_Retry/lease-type=LeaseLeader
--- FAIL: TestReplicaCircuitBreaker_Partial_Retry/lease-type=LeaseLeader (10.42s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv-triage
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReplicaCircuitBreaker_Partial_Retry.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-54774

--- a/pkg/kv/kvserver/client_replica_circuit_breaker_test.go
+++ b/pkg/kv/kvserver/client_replica_circuit_breaker_test.go
@@ -793,6 +793,11 @@ func TestReplicaCircuitBreaker_Partial_Retry(t *testing.T) {
 	skip.UnderRace(t)
 	skip.UnderDeadlock(t)
 
+	// To help debug issues like #154179.
+	prevVModule := log.GetVModule()
+	defer func() { _ = log.SetVModule(prevVModule) }()
+	require.NoError(t, log.SetVModule("dist_sender=3"))
+
 	testutils.RunValues(t, "lease-type", roachpb.ExpirationAndLeaderLeaseType(),
 		func(t *testing.T, leaseType roachpb.LeaseType) {
 			// Use a context timeout, to prevent test hangs on failures.

